### ISSUE TITLE
handle err on marshal instead of panic

### ISSUE THREAD

**Issue by poonai** [sch00lb0y/vegamcache#8]

**Comment by sidthekidder** [comment]
Do you mean something like this? 
```
func marshal(data interface{}) []byte {
        marshaledData, err := json.Marshal(data)
        if err != nil {
-               panic(err)
+               return []byte("502 Server Error")
        }
        return marshaledData
 }
```

**Comment by poonai** [comment]
marshal should return marshaledData and err. if err, response writer should write `"502 Server Error"`

**Comment by sidthekidder** [comment]
Yes that's what will happen right, we are shifting the err check into marshal() func. If its successful marshaledData will get returned and written, else "server error" bytes will get written. Otherwise the err check will get replicated 3 times. 

How do I test this error?

**Comment by poonai** [comment]
to get err, pass a something which not a json.
>>> err check will get replicated 3 times 

It looks idiomatic, that's why to return err
